scripts/zones/Southern_San_dOria/npcs/Kipopo.lua

```diff
@@ -2,38 +2,77 @@
 -- Area: Southern San d'Oria
 --  NPC: Kipopo
 -- Type: Leathercraft Synthesis Image Support
--- !pos -191.050 -2.15 12.285 230
+-- !pos  -191.050 -2.15 12.285 230
 -----------------------------------
-require("scripts/globals/status");
-require("scripts/globals/crafting");
-local ID = require("scripts/zones/Southern_San_dOria/IDs");
+require("scripts/globals/status")
+require("scripts/globals/crafting")
+require("scripts/globals/quests")
+require("scripts/globals/npc_util")
+local ID = require("scripts/zones/Southern_San_dOria/IDs")
 -----------------------------------
 
 function onTrade(player,npc,trade)
+    local sayItWithAHandbagCS = player:getVar("sayItWithAHandbagCS")
+
+    if player:hasKeyItem(dsp.ki.TORN_PATCHES_OF_LEATHER) and sayItWithAHandbagCS == 2 then
+        if npcUtil.tradeHasExactly(trade, {2012, 850, 816}) then
+            player:startEvent(910)
+        end
+    end
 end;
 
 function onTrigger(player,npc)
-    local guildMember = isGuildMember(player,7);
-    local SkillCap = getCraftSkillCap(player, dsp.skill.LEATHERCRAFT);
-    local SkillLevel = player:getSkillLevel(dsp.skill.LEATHERCRAFT);
+    local sayItWithAHandbag = player:getQuestStatus(CRYSTAL_WAR, dsp.quest.id.crystalWar.SAY_IT_WITH_A_HANDBAG)
+    local sayItWithAHandbagCS = player:getVar("sayItWithAHandbagCS")
+    local sayItWithAHandbagBonusCS = player:getVar("sayItWithAHandbagBonusCS")
+    local guildMember = isGuildMember(player,7)
+    local SkillCap = getCraftSkillCap(player, dsp.skill.LEATHERCRAFT)
+    local SkillLevel = player:getSkillLevel(dsp.skill.LEATHERCRAFT)
 
-    if (guildMember == 1) then
-        if (player:hasStatusEffect(dsp.effect.LEATHERCRAFT_IMAGERY) == false) then
-            player:startEvent(651,SkillCap,SkillLevel,1,239,player:getGil(),0,0,0);
+    if sayItWithAHandbag == QUEST_COMPLETED and player:hasItem(19110) and sayItWithAHandbagBonusCS == 1 then
+        player:startEvent(914)
+    elseif player:hasKeyItem(dsp.ki.REPAIRED_HANDBAG) and sayItWithAHandbagCS == 4 then
+        player:startEvent(913)
+    elseif sayItWithAHandbagCS == 3 then
+        if player:needToZone() then
+            player:startEvent(911)
         else
-            player:startEvent(651,SkillCap,SkillLevel,1,239,player:getGil(),7128,0,0);
+            player:startEvent(912)
+        end
+    elseif sayItWithAHandbagCS == 2 then
+        player:startEvent(909)
+    elseif player:hasKeyItem(dsp.ki.TORN_PATCHES_OF_LEATHER) and sayItWithAHandbagCS == 1 then
+        player:startEvent(908)
+    elseif guildMember == 1 then
+        if player:hasStatusEffect(dsp.effect.LEATHERCRAFT_IMAGERY) == false then
```

Comment:
If not player:hasstatus....
Remove the == false

Reply:
@KnowOne134 I dunno if that will break imagery, I haven't touched imagery  😿️ dunno why it shows that as a change.

Reply:
Can change old code to new style as we edit on is all

Reply:
Will do my best, will make changes and test 😸️